lib/HLSL/DxilContainerAssembler.cpp

```diff
@@ -135,8 +135,10 @@ struct sort_second {
 struct sort_sig {
   bool operator()(const DxilProgramSignatureElement &a,
                   const DxilProgramSignatureElement &b) {
-    return (a.Stream < b.Stream) |
-           ((a.Stream == b.Stream) & (a.Register < b.Register));
+    return (a.Stream < b.Stream) ||
+           ((a.Stream == b.Stream) && (a.Register < b.Register)) ||
+           ((a.Stream == b.Stream) && (a.Register == b.Register) &&
+            (a.SemanticName < b.SemanticName));
```

Comment:
loathe as I am to add friction to the introduction of this change, I feel I have to disclose that this might require more scrutiny than it seems. My first crack at this ordered by the mask. That failed because the signatures didn't match those produced by dxil.dll. I have no access to the implementation of dxil.dll. In my nightmares, I imagine that it relies entirely on the arbitrary behavior of std::sort as this did previously. That seems like a fact worth knowing.

Reply:
+1. Based on the definition of ``DxilProgramSignatureElement``, I think this is a more accurate implementation.

Reply:
Yes, this change is probably the best to match existing dxil.dll.